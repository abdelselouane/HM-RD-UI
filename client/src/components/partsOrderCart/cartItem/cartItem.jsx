import React from 'react';
import PropTypes from 'prop-types';

import Counter from '../../counter';
import { withContext } from '../../../contexts/orderParts';
import { decodeHtml } from '../../../utils/formatHelper';

export const CartItem = ({
  handleRemoveFromCart,
  handleAdjustQuantity,
  ...part
}) => (
  <div className="cart-item">
    <div className="part-info">
      <h3 className="part-desc">{decodeHtml(part.partDescription)}</h3>
      <span className="part-nbr">{part.partNbr}</span>
    </div>
    <div className="qty-counter counter-input sm">
      <span className="qty-label" htmlFor={`${part.partNbr}-qty`}>
        Qty
      </span>
      <Counter
        name={part.partNbr}
        value={part.quantity}
        handleChangeValue={handleAdjustQuantity}
        readOnly
      />
    </div>
    <div
      className="remove-btn"
      onClick={() => handleRemoveFromCart(part)}
    >
      <span>Remove</span>
      <i className="icon_trash" />
    </div>
  </div>
);

CartItem.propTypes = {
  handleAdjustQuantity: PropTypes.func.isRequired,
  handleRemoveFromCart: PropTypes.func.isRequired,
  partNbr: PropTypes.string.isRequired,
  partDescription: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired
};

export default withContext(CartItem);
