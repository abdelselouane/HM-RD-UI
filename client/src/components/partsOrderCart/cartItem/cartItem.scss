.cart-item {
  display: flex;
  width: 94%;
  margin: 0 3%;
  border-bottom: solid 2px $black12;
  padding: 1em 0em;

  .part-info {
    flex: 0.4;
    display: flex;
    flex-direction: column;

    .part-desc,
    .part-nbr {
      font-size: 20px;
    }
  }

  .qty-counter {
    flex: 0.45;
    flex-direction: row;
    align-items: center;
    justify-content: center;

    .qty-label {
      margin-right: 0.5em;
      font-weight: 900;
    }
  }

  .remove-btn {
    cursor: pointer;
    color: $orange;
    flex: 0.15;
    text-align: center;
    display: flex;
    align-items: center;
    font-weight: 900;

    i.icon_trash {
      font-size: 1.75em;
    }
  }
}
