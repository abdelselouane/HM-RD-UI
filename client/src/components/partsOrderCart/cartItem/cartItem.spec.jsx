/* global beforeAll, beforeEach, describe, expect, it, jest */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { CartItem } from './cartItem';
import Counter from '../../counter';

describe('<CartItem />', () => {
  let wrapper;
  let spy;
  let props;
  let part;

  beforeAll(() => {
    part = {
      partNbr: 'mockPartNumber',
      partDescription: '&amp; mockDescription',
      brandName: 'mockbrandName',
      quantity: 1
    };

    spy = null;
  });

  beforeEach(() => {
    props = {
      ...part,
      handleAdjustQuantity: jest.fn(),
      handleRemoveFromCart: jest.fn()
    };

    wrapper = shallow(<CartItem {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render escaped part partDescription', () => {
    expect(wrapper.find('.part-desc').text()).toBe('& mockDescription');
  });

  it('should render part number', () => {
    expect(wrapper.find('.part-nbr').text()).toBe('mockPartNumber');
  });

  it('should render qty label', () => {
    expect(wrapper.find('.qty-label').text()).toBe('Qty');
  });

  it('should render Counter', () => {
    expect(wrapper.find(Counter)).toHaveLength(1);
  });

  it('should render remove button with label', () => {
    expect(wrapper.find('.remove-btn').text()).toBe('Remove');
    expect(wrapper.find('.remove-btn .icon_trash')).toHaveLength(1);
  });

  it('should call handleRemoveFromCart when remove-btn is clicked', () => {
    spy = jest.spyOn(props, 'handleRemoveFromCart');
    wrapper.find('.remove-btn').simulate('click');
    expect(spy).toHaveBeenCalledWith(part);
  });
});
