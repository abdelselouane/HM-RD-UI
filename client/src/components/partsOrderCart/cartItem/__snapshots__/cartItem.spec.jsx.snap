// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> should render properly 1`] = `
<div
  className="cart-item"
>
  <div
    className="part-info"
  >
    <h3
      className="part-desc"
    >
      & mockDescription
    </h3>
    <span
      className="part-nbr"
    >
      mockPartNumber
    </span>
  </div>
  <div
    className="qty-counter counter-input sm"
  >
    <span
      className="qty-label"
      htmlFor="mockPartNumber-qty"
    >
      Qty
    </span>
    <Styled(Counter)
      handleChangeValue={[MockFunction]}
      name="mockPartNumber"
      readOnly={true}
      value={1}
    />
  </div>
  <div
    className="remove-btn"
    onClick={[Function]}
  >
    <span>
      Remove
    </span>
    <i
      className="icon_trash"
    />
  </div>
</div>
`;
